File: Ruby/lib/civitas-ruby/juego_texto.rb
# encoding:utf-8
require_relative 'civitas_juego'
require_relative 'vista_textual'
require_relative 'dado'
require_relative 'controlador'

#TODO: Arreglar visibilidad de métodos a partir de jugador.rb
module Civitas
  class JuegoTexto

    def initialize
      @vista = Vista_textual.new
      nro_jugadores = @vista.lee_entero(4, "Indique el número de jugadores: ", "Valor incorrecto. Solo pueden jugar hasta 4 jugadores")
      nombres = []
      puts "Indique el nombre de los jugadores: "
      for i in 0..nro_jugadores-1
        nombres << gets.chomp
      end
      @juego = CivitasJuego.new(nombres)
      #(["Alejandro", "Brutus", "Camila", "Diana"])
      Dado.instance.debug = true
      @controlador = Controlador.new(@juego, @vista)
      @controlador.juega
    end
      
  end
end

Civitas::JuegoTexto.new

File: juegoTexto/JuegoTexto.java
package juegoTexto;

import civitas.CivitasJuego;
import java.util.ArrayList;
import GUI.Dado;
import java.util.Scanner;

class JuegoTexto {
  public static void main(String[] args) {
    VistaTextual vista = new VistaTextual();
    int nroJugadores = vista.leeEntero(4, "Indique el número de jugadores: ", "Valor incorrecto. Solo pueden jugar hasta 4 jugadores");
    
    ArrayList<String> nombres = new ArrayList<>(nroJugadores);
    System.out.println("Introduce el nombre de los jugadores: ");
    Scanner in = new Scanner(System.in);
    for (int i = 0; i < nroJugadores; i++) {
      nombres.add(in.nextLine());
    }
    // in.close();
    // nombres.add("Alejandro");
    // nombres.add("Brutus");
    // nombres.add("Camila");
    // nombres.add("Diana");
    CivitasJuego juego = new CivitasJuego(nombres);
    Dado.getInstance().setDebug(true);
    Controlador controlador = new Controlador(juego, vista);
    controlador.juega();
    in.close();
  }
}
